packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2291,6 +2293,9 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       allowsDirectSuper: boolean,
     ): void {
       const typeParameters = this.tsTryParseTypeParameters();
+      if (typeParameters && isConstructor) {
+        this.raise(method.start, TSErrors.ConstructorHasTypeParameters);
```

Comment:
It looks like TS reports this error on the [generic declaration](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDC0G8BQ1rAPYDsIBcBOArsDungDwAqAfABQBm66AXNBQJSIC+SnQA). Should we try to match the error location?

Reply:
Thank you. I've fixed:smile: 952ef30